feat: add file type variables to bundles (#631)
defenseunicorns/uds-cli: docs/overrides.md

1. `uds-config.yaml` variables
1. Variables `default` in the`uds-bundle.yaml`

#### Variable Types
Variables can be of either type `raw` or `file`. The type will default to raw if not set explicitly.

> [!WARNING]  
> If a variable is set to accept a file as it's value, but is missing the `file` type, then the file will not be processed.

```yaml
kind: UDSBundle
metadata:
   name: example-bundle
   version: 0.0.1

packages:
   - name: helm-overrides-package
     path: "../../packages/helm"
     ref: 0.0.1
     overrides:
        podinfo-component:
          unicorn-podinfo:
           variables:
           - name: test_secret
             path: "testSecret"
             description: "Key to be set in a secret"
             type: file

> ```

Reviewer: Love the example! Let's be a bit more explicit about what `raw` vs `file` means. Recommend adding another variable to the list of `variables` and say explicitly what each will resolve to (ie. variable `test-secret` will resolve to the contents of the file `testSecret` <- even better if we make this a more realistic-looking file with an actual extension)

Author: added a "side by side" raw example and some descriptions to more explicitly detail what's going on. let me know what you think.

**File Paths**

If a file path is not absolute it will be set as relative to the uds-config.yaml path.

e.g. the following `uds-config.yaml` is in [`src/test/bundles/07-helm-overrides/variable-files/`](../src/test/bundles/07-helm-overrides/variable-files/uds-config.yaml)
```yaml
variables:
  helm-overrides:
    test_secret: test.pub
```

This means when `test.pub` is evalutated it will first be appended to the config path like so `src/test/bundles/07-helm-overrides/variable-files/test.pub`.

If the file path is already set to the same relative path as the config, then no merging will take place.

> [!NOTE]  

> > uds-cli does not encrypt or base64 encode any file contents before passing said data to Zarf or Helm.  

Reviewer: For consistency, let's capitalize "UDS CLI" and remove the `-`

> > So if for example the file contains a key to be used in a kubernetes secret, it must be base64 encoded before given to uds-cli.

Reviewer: ```suggestion
> For example, if the file contains a key to be used in a Kubernetes secret, it must be base64 encoded before being ingested by UDS CLI.
```
